debezium-connector-oracle/src/test/java/io/debezium/connector/oracle/OracleConnectorIT.java

```diff
@@ -1642,6 +1642,84 @@ public void shouldContainPartitionInSchemaChangeEvent() throws Exception {
         }
     }
 
+    @Test
+    @FixFor("DBZ-3322")
+    public void shouldNotEmitEventsOnConstraintViolations() throws Exception {
+        TestHelper.dropTable(connection, "dbz3322");
+        try {
+            connection.execute("CREATE TABLE dbz3322 (id number(9,0), data varchar2(50))");
+            connection.execute("CREATE UNIQUE INDEX uk_dbz3322 ON dbz3322 (id)");
+            TestHelper.streamTable(connection, "dbz3322");
+
+            Configuration config = TestHelper.defaultConfig()
+                    .with(OracleConnectorConfig.TABLE_INCLUDE_LIST, "DEBEZIUM\\.DBZ3322")
+                    .build();
+
+            start(OracleConnector.class, config);
+            assertConnectorIsRunning();
+
+            waitForStreamingRunning(TestHelper.CONNECTOR_NAME, TestHelper.SERVER_NAME);
+
+            try {
+                connection.executeWithoutCommitting("INSERT INTO dbz3322 (id,data) values (1, 'Test')");
+                connection.executeWithoutCommitting("INSERT INTO dbz3322 (id,data) values (1, 'Test')");
```

Comment:
How about we use two distinct values for the two rows, so to assert on that value that's actually coming through?